### PR TITLE
Add argument to drop column in ColumnFormula

### LINKED ISSUE #508: Improve error message for `ColumnFormula` constraint when constraint column used in formula
Using the `ColumnFormula` constraint where the constraint column is used in the constraint formula crashes with a `KeyError`. While this is expected, the error message could be improved to provide a more meaningful explanation of what's causing the issue. 

Below is some code to produce this error:
```python3
import pandas as pd
from sdv.constraints import *
from sdv.tabular import *

ones = [1 for i in range(10)]
data = pd.DataFrame({
    'a': ones,
    'b': ones,
    'c': ones
})

def formula(data):
    return data['a']

constraint = ColumnFormula(
    column='a',
    formula=formula
)

model = GaussianCopula(constraints=[constraint])
model.fit(data)
samples = model.sample(10)
```

### PR DESCRIPTION
**Pull request by katxiao**
Resolves #508

### CODE REVIEW

File: tests/unit/constraints/test_tabular.py [hunk 1]
```diff
@@ -1802,6 +1802,36 @@ def test_transform(self):
...
 
+    def test_transform_without_dropping_column(self):
+        """Test the ``ColumnFormula.transform`` method without dropping the column.
+
+        If `drop_column` is false, expect to not drop the constraint column.
+
+        Input:
+        - Table data (pandas.DataFrame)
+        Output:
+        - Table data without the indicated column (pandas.DataFrame)
```
Comment (amontanez24) [inline tests/unit/constraints/test_tabular.py:1813]: This should say "with the indicated column" right? Since it isn't being dropped

**Review by amontanez24 — APPROVED** [review]
One small comment but besides that LGTM!

File: tests/unit/constraints/test_tabular.py (hunk 1 above)
Comment (katxiao) [reply tests/unit/constraints/test_tabular.py:1813]: Yes, thanks!

**Review by csala — APPROVED** [review]
LGTM!

**Pull request merged**